### PR TITLE
Make multi-participant conformance tests work with distributed committers/event logs [KVL-1335]

### PR DESCRIPTION
**Pull request by hubert-da**
Uncovered when creating per-participant event logs for the PKV reference implementation and tested with it.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Synchronize.scala [hunk 1]
```diff
@@ -34,13 +29,6 @@ object Synchronize {
       bob <- beta.allocateParty()
       _ <- alpha.waitForParties(Set(beta), Set(alice, bob))
       _ <- beta.waitForParties(Set(alpha), Set(alice, bob))
-      factory <- alpha.create(alice, AgreementFactory(bob, alice))
```
Comment (hubert-da) [inline ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Synchronize.scala:37]: @digital-asset/kv-participant Does this simplification make sense to you?

Comment (nicu-da) [reply ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Synchronize.scala:37]: @hubert-da good point, I think it makes sense as both participants see the newly allocated parties therefore they should have any contracts created before this call. Let's see what the participant team thinks.

File: ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceCorrectnessIT.scala [hunk 2]
```diff
@@ -154,6 +156,7 @@ class TransactionServiceCorrectnessIT extends LedgerTestSuite {
     "TXTransactionTreeByIdSameAsTransactionStream",
     "Expose the same events for each transaction as the output of getTransactionTrees",
     allocate(SingleParty, SingleParty),
+    enabled = _.committerEventLog.eventLogType.isCentralized,
```
Comment (nicu-da) [inline ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceCorrectnessIT.scala:159]: Why is this test enabled only for centralized committers?

Comment (hubert-da) [reply ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceCorrectnessIT.scala:159]: Similarly, due to potentially different transaction IDs and `flatTransactionById`. An alternative is to search for a contract id as we do in some tests. I didn't want to switch all of the tests to leave coverage of by-id queries.

File: ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Synchronize.scala (hunk 1 above)
Comment (tudor-da) [reply ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Synchronize.scala:37]: I think that's sane as well.

**Review by tudor-da — APPROVED** [review]

File: ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceCorrectnessIT.scala [hunk 3]
```diff
@@ -312,35 +316,61 @@ class TransactionServiceCorrectnessIT extends LedgerTestSuite {
...
+          events = t.events
+            .map(e => e.copy(event = stripEventId(e.event)).modifyWitnessParties(_.sorted)),
+          transactionId = "transactionId",
         )
       )
-      .sortBy(_.transactionId)
+  }
 
-  // Strip command id and offset to yield a transaction comparable across participant
+  // Strip command id, offset, event id and transaction id to yield a transaction comparable across participant
```
Comment (miklos-da) [inline ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceCorrectnessIT.scala:344]: Nit: `across participants`

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/LedgerFeatures.scala [hunk 4]
```diff
@@ -13,4 +15,6 @@ case class LedgerFeatures(
     commandDeduplicationFeatures: CommandDeduplicationFeatures =
       CommandDeduplicationFeatures.defaultInstance,
     contractIdFeatures: ExperimentalContractIds = ExperimentalContractIds.defaultInstance,
+    committerEventLog: ExperimentalCommitterEventLog =
```
Comment (miklos-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/LedgerFeatures.scala:18]: Nit: You could simply use `ExperimentalCommitterEventLog.defaultInstance` here for consistency.

File: ledger/ledger-on-memory/src/app/scala/com/daml/ledger/on/memory/Owner.scala [hunk 5]
```diff
@@ -3,17 +3,48 @@
...
 import com.daml.ledger.resources.ResourceOwner
+import com.daml.platform.apiserver.LedgerFeatures
+import com.daml.platform.services.time.TimeProviderType
 
 object Owner {
-  // Utily if you want to spin this up as a library.
-  def apply(config: Config[Unit]): ResourceOwner[Unit] =
+  // Utility if you want to spin this up as a library.
+  def apply(config: Config[Unit]): ResourceOwner[Unit] = {
+    val features = LedgerFeatures(
```
Comment (miklos-da) [inline ledger/ledger-on-memory/src/app/scala/com/daml/ledger/on/memory/Owner.scala:21]: For better visibility and readability, you might want to move the definition of this `LedgerFeatures` into a `def reasonableDefaultsForLedgerFeatures` or similar (alternatively, you could make it a constant if you overwrite the `staticTime` field here).

**Pull request merged**